### PR TITLE
gh-101326: Fix regression when passing None to FutureIter.throw

### PR DESCRIPTION
**Pull request by hauntsaninja**
* Issue: gh-101326

### CODE REVIEW

File: Misc/NEWS.d/next/Library/2023-01-25-18-07-20.gh-issue-101326.KL4SFv.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix regression when passing ``None`` to ``FutureIter.throw``.
```
Comment (srittau) [inline Misc/NEWS.d/next/Library/2023-01-25-18-07-20.gh-issue-101326.KL4SFv.rst]:
```suggestion
Fix regression when passing ``None`` as third argument to ``FutureIter.throw``.
```

**Comment by srittau** [comment]
The windows test failure seems unrelated.

**Review by gvanrossum — APPROVED** [review]
Thanks, that definitely happened by accident.

**Comment by hauntsaninja** [comment]
The update to the news entry isn't right. There is also a regression when passing None to the second argument.

File: Misc/NEWS.d/next/Library/2023-01-25-18-07-20.gh-issue-101326.KL4SFv.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Fix regression when passing ``None`` as third argument to ``FutureIter.throw``.
```
Comment (hauntsaninja) [inline Misc/NEWS.d/next/Library/2023-01-25-18-07-20.gh-issue-101326.KL4SFv.rst]:
```suggestion
Fix regression when passing ``None`` as second or third argument to ``FutureIter.throw``.
```

**Review by gvanrossum — APPROVED** [review]

**Comment by hauntsaninja** [comment]
`OSError: [Errno 28] No space left on device` <-- going to assume buildbot failure is unrelated to this PR and not take further action

**Pull request merged**